DuckDuckGo/OnboardingExperiment/OnboardingIntro/OnboardingView.swift

```diff
@@ -82,17 +88,25 @@ struct OnboardingView: View {
             }
             .frame(width: geometry.size.width, alignment: .center)
             .offset(y: geometry.size.height * Metrics.dialogVerticalOffsetPercentage.build(v: verticalSizeClass, h: horizontalSizeClass))
+            .onAppear {
+                DispatchQueue.main.asyncAfter(deadline: .now() + Metrics.daxDialogVisibilityDelay) {
```

Comment:
Looks like that even if I use `withAnimation(.linear.delay(0.5)) {}` the content of the closure is not executed with a delay but immediately, hence the text appears as it is already been tapped.

Reply:
The way to do this would be to change `AnimatableTypingText` view to support the` Animatable` protocol which tbh I should have caught early on when you started this, sorry.

Would need a bit of a refactor because passing in true to that class starts a timer, but you don't need any of that if you use Animatable - just need to know what value you are interpolating to, so rather than starting a timer you have your start index at zero and target index at the length of the string and let SwiftUI interpolate between them for you using the `animatableData` property required by the protocol (an empty default is provided).

In places you don't use `withAnimation` then the text will just appear in full by default, so kind of the opposite of what's happening here.

It's another one of those 'mindset' changes :) 

It's also fairly advanced topic so we can skip it for now, I think.  It'd be nice to do it "right" at some point though.

Reply:
Hey @brindy no worries and thanks for pointing this out. Would like to give it a go when I have some down time!